store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -513,4 +513,36 @@ private void makeAllGraphsFromCache() throws StorageException {
             }
         }
     }
+
+    HashMap<String, Object> getAllGraphsAndAuths(final User user, final List<String> graphIds) {
+        final HashMap<String, Object> graphIdFedAccessMap = new HashMap<>();
+        storage.entrySet()
+                .stream()
+                .filter(entry -> isValidToView(user, entry.getKey()))
+                .forEach(entry -> {
+                    FederatedAccess federatedAccess = entry.getKey();
+                    populateGraphIdAccessMap(graphIdFedAccessMap, graphIds, federatedAccess, entry.getValue());
+                });
+
+        return graphIdFedAccessMap;
+    }
+
+    private void populateGraphIdAccessMap(final HashMap<String, Object> graphIdFedAccessMap, final List<String> graphIds, final FederatedAccess federatedAccess, final Set<Graph> graphs) {
+        for (final Graph g : graphs) {
+            if (nonNull(graphIds) && !graphIds.isEmpty()) {
```

Comment:
Don't think you need this second check

Reply:
As we have seen this is to change the logic.
if no lists is provided I want all the graphs. I'll add text.